src/test/app/PermissionedDomains_test.cpp

```diff
@@ -0,0 +1,478 @@
+        uint256 domain2;
+        {
+            BEAST_EXPECT(
+                credentials10.size() == PermissionedDomainSet::PD_ARRAY_MAX);
+            BEAST_EXPECT(credentials10 != sortCredentials(credentials10));
+            env(setTx(alice, credentials10), fee(setFee));
+            auto tx = env.tx()->getJson(JsonOptions::none);
+            auto meta = env.meta()->getJson(JsonOptions::none);
+            Json::Value a(Json::arrayValue);
+            a = meta["AffectedNodes"];
+
+            for (auto const& node : a)
+            {
+                if (!node.isMember("CreatedNode") ||
+                    node["CreatedNode"]["LedgerEntryType"] !=
+                        "PermissionedDomain")
+                {
+                    continue;
+                }
+                std::ignore = domain2.parseHex(
+                    node["CreatedNode"]["LedgerIndex"].asString());
+            }
+            auto objects = getObjects(alice, env);
+            auto object = objects[domain2];
+            BEAST_EXPECT(
+                credentialsFromJson(object) == sortCredentials(credentials10));
+        }
+
+        // Make a new domain with insufficient fee.
+        env(setTx(alice, credentials10),
+            fee(drops(dropsFee - 1)),
+            ter(telINSUF_FEE_P));
+
+        // Update with 1 credential.
+        env(setTx(alice, credentials1, domain2));
+        BEAST_EXPECT(
+            credentialsFromJson(getObjects(alice, env)[domain2]) ==
+            credentials1);
+
+        // Update with 10 credentials.
+        env(setTx(alice, credentials10, domain2));
+        env.close();
+        BEAST_EXPECT(
+            credentialsFromJson(getObjects(alice, env)[domain2]) ==
+            sortCredentials(credentials10));
+
+        // Test bad data when creating a domain.
+        testBadData(alice, env);
+        // Test bad data when updating a domain.
+        testBadData(alice, env, domain2);
+
+        // Try to delete the account with domains.
+        auto const acctDelFee(drops(env.current()->fees().increment));
+        constexpr std::size_t deleteDelta = 255;
+        {
+            // Close enough ledgers to make it potentially deletable if empty.
+            std::size_t ownerSeq = ownerInfo(alice, env)["Sequence"].asUInt();
+            while (deleteDelta + ownerSeq > env.current()->seq())
+                env.close();
+            env(acctdelete(alice, alice2),
+                fee(acctDelFee),
+                ter(tecHAS_OBLIGATIONS));
+        }
+
+        {
+            // Delete the domains and then the owner account.
+            for (auto const& objs : getObjects(alice, env))
+                env(deleteTx(alice, objs.first));
+            env.close();
+            std::size_t ownerSeq = ownerInfo(alice, env)["Sequence"].asUInt();
+            while (deleteDelta + ownerSeq > env.current()->seq())
+                env.close();
+            env(acctdelete(alice, alice2), fee(acctDelFee));
+        }
+    }
```

Comment:
Please add tests form codecov:
DomainID has invalid data (not uint256)
DomainID object doesn't exists
Other account(not owner) try to update existing domain

Reply:
I added a test for a non-owner updating an existing domain.
I removed the check for DomainID object not existing, because it's a required field and is already checked while parsing the input.
DomainID has to be uint256 or the transaction won't parse, so that's an unnecessary check.